contracts/ensWildcardResolvers/DigidaigakuResolver.sol

```diff
@@ -0,0 +1,148 @@
+        bytes calldata parentDnsEncoded
+    ) internal view virtual override returns (address) {
+        // Extract tokenId from child name
+        (bool valid, uint256 tokenId) = childUtf8Encoded.parseUint256();
+        // No token, try resolving using name
+        if (!valid) {
+            string[] memory urls = new string[](1);
+            urls[0] = url;
+            bytes memory extraData = abi.encode(parentDnsEncoded);
+
+            revert OffchainLookup(
+                address(this),
+                urls,
+                childUtf8Encoded,
+                this.resolveByNameCallback.selector,
+                extraData
+            );
+        }
+
+        return resolveOwnerAddress(tokenId);
+    }
+
+    function resolveByNameCallback(
+        bytes calldata response,
+        bytes calldata extraData
+    ) public view returns (bytes memory) {
+        // Get tokenId from offchain response
+        uint256 tokenId = abi.decode(response, (uint256));
+        return abi.encode(resolveOwnerAddress(tokenId));
+    }
+
+    function resolveOwnerAddress(uint256 tokenId) internal view returns (address) {
+        // Lookup token owner
+        address tokenOwner;
+        try digidaigakuContract.ownerOf(tokenId) returns (address _tokenOwner) {
+            tokenOwner = _tokenOwner;
+        } catch {
+            tokenOwner = address(0);
+        }
+        return tokenOwner;
+    }
+
+    function _resolveWildcardTextRecord(
+        bytes calldata childUtf8Encoded,
+        bytes calldata parentDnsEncoded,
+        string calldata key
+    ) internal view virtual override returns (string memory) {
+        // Extract tokenId from child name
+        (bool valid, uint256 tokenId) = childUtf8Encoded.parseUint256();
+        // No token, try resolving using name
+        if (!valid) {
+            string[] memory urls = new string[](1);
+            urls[0] = url;
+
+            revert OffchainLookup(
+                address(this),
+                urls,
+                childUtf8Encoded,
+                this.resolveTextByNameCallback.selector,
+                abi.encode(key)
+            );
+        }
+
+        return resolveTextRecord(key, tokenId);
+    }
+
+    function resolveTextByNameCallback(
+        bytes calldata response,
+        bytes calldata extraData
+    ) public view returns (bytes memory) {
+        // Get tokenId from offchain response
+        uint256 tokenId = abi.decode(response, (uint256));
+        // Get text key from extraData
+        string memory key = abi.decode(extraData, (string));
+        return abi.encode(resolveTextRecord(key, tokenId));
+    }
+
+    function resolveTextRecord(string memory key, uint256 tokenId) internal view returns (string memory) {
+        // Don't bother returning anything if this tokenId has never been minted
+        // solhint-disable-next-line no-empty-blocks
+        try digidaigakuContract.ownerOf(tokenId) {} catch {
+            return "";
+        }
```

Comment:
Why doesn't this use `resolveOwnerAddress()`?

Reply:
It doesn't need to resolve the address, it just needs to verify this tokenId exists and it owned

Reply:
Ah yup! Sorry